### PR TITLE
Allow hash checkers to be configured

### PR DESCRIPTION
**Pull request by tiagovtristao**

### CODE REVIEW

File: src/build/build_step_test.go [hunk 1]
```diff
@@ -507,6 +530,25 @@ func TestSha1SingleHash(t *testing.T) {
 	}
 }
 
+func newStateWithHashCheckers(label, hashFunction string, hashCheckers []string) (*core.BuildState, *core.BuildTarget) {
+	config, _ := core.ReadConfigFiles(nil, nil)
+	if len(hashFunction) > 0 {
```
Comment (peterebden) [inline src/build/build_step_test.go:535]: is this really what you meant given this is a `string` (but `hashCheckers` is a `[]string`)?

File: src/build/build_step_test.go [hunk 2]
```diff
@@ -507,6 +530,25 @@ func TestSha1SingleHash(t *testing.T) {
 	}
 }
 
+func newStateWithHashCheckers(label, hashFunction string, hashCheckers []string) (*core.BuildState, *core.BuildTarget) {
```
Comment (peterebden) [inline src/build/build_step_test.go:533]: this might be a good case for a `...string`?

**Review by peterebden — APPROVED** [review]

File: src/build/build_step_test.go (hunk 2 above)
Comment (tiagovtristao) [reply src/build/build_step_test.go:533]: Yeah, I guess it a little nicer.

File: src/build/build_step_test.go (hunk 1 above)
Comment (tiagovtristao) [reply src/build/build_step_test.go:535]: Funny enough I thought about that, but the end result should be the same (as opposed to `hashFunction != ""`). I guess, idiomatically speaking and for consistency `!= ""` is more in line with checking empty strings. Happy to change.

**Review by Tatskaari — APPROVED** [review]

**Pull request merged**